docker/src/main/java/org/jclouds/docker/compute/strategy/DockerComputeServiceAdapter.java

```diff
@@ -136,6 +140,14 @@ public DockerComputeServiceAdapter(DockerApi api) {
               .publishAllPorts(true)
               .privileged(true);
 
+      if (templateOptions.getDirectPorts().isPresent()) {
+         Map<String, List<Map<String, String>>> portBindings = Maps.newHashMap();
+         for (Integer port : templateOptions.getDirectPorts().get()) {
+            portBindings.put(port + "/tcp", Lists.<Map<String, String>>newArrayList(ImmutableMap.of("HostPort", Integer.toString(port))));
```

Comment:
It may not have been specified there, though. For example, a Dockerfile will have `EXPOSE` commands, but jclouds does not need to know about them, Docker will find out when it starts the container...